src/pl/senla/hotel/repository/RepositoryOrderCollection.java

```diff
@@ -44,4 +46,14 @@ public boolean delete(int id) {
         readAll().remove(id);
         return true;
     }
+
+    @Override
+    public List<HotelService> readAllServicesSortByPrice(int idGuest) {
+        return readAll().stream().filter(hs -> hs.getClient().getIdGuest() == idGuest).flatMap(o -> o.getServices().stream()).sorted(new HotelServicesComparatorByPrice()).toList();
```

Comment:
This task is in work. So I haven't still pushed it.

Perhaps there will be some questions because HotelService is an abstract class and it has only 1 attribute inside (Guest guest).
Price and Dates/Times I moved to its child classe. May be it's mistake... And it is better to return these attributes back to parent class.

I wrote such Comparator:
public class HotelServicesComparatorByPrice implements Comparator<HotelService> {
    @Override
    public int compare(HotelService o1, HotelService o2) {
        switch(o1.getTypeOfService()){
            case "RoomReservation":
                return compareRoomReservation((RoomReservation) o1, (RoomReservation) o2);
            case "Restaurant":
                return compareRestaurant((Restaurant) o1, (Restaurant) o2);
            case "GuideTour":
                return compareGuideTour((GuideTour) o1, (GuideTour) o2);
            default:
                System.out.println("ERROR IN SERVICE TYPE.");
        }
        return 0;
    }

    private int compareRoomReservation(RoomReservation o1, RoomReservation o2){
        return o1.getRoom().getRoomPrice() - o2.getRoom().getRoomPrice();
    }

    private int compareRestaurant(Restaurant o1, Restaurant o2){
        return o1.getPrice() - o2.getPrice();
    }

    private int compareGuideTour(GuideTour o1, GuideTour o2){
        return o1.getPrice() - o2.getPrice();
    }
}

It looks too "heavy" but it works (tested with RoomReservations only).